src/store.test.ts

```diff
@@ -0,0 +1,117 @@
+import {
+    registerStateChangeHandler,
+    getConsentState,
+    setConsentState,
+    _,
+} from './store';
+import { readIabCookie, readLegacyCookie } from './cookies';
+
+jest.mock('./cookies', () => ({
+    readGuCookie: jest.fn(),
+    readIabCookie: jest.fn(),
+    readLegacyCookie: jest.fn(),
+}));
+
+const myCallBack = jest.fn();
+
+const guDefaultState = { functional: true, performance: true };
+const iabDefaultState = { 1: null, 2: null, 3: null, 4: null, 5: null };
+const guMixedState = { functional: true, performance: false };
+const iabMixedState = { 1: true, 2: false, 3: false, 4: true, 5: true };
+const iabMixedString =
+    'BOpj_ITOpj_ITBwABAENCtmAAAAsJ7_______9______9uz_Ov_v_f__33e8__9v_l_7_-___u_-3zd4u_1vf99yfm1-7etr3tp_87ues2_Xur__79__3z3_9phP78k89r7337Ew-v-3o8bzBAI';
+const legacyTrueCookie = '1.54321';
+const iabTrueState = { 1: true, 2: true, 3: true, 4: true, 5: true };
+
+describe('Store', () => {
+    beforeAll(() => {});
+    afterEach(() => {
+        jest.resetAllMocks();
+        _.resetModule();
+    });
+
+    describe('sets the state correctly', () => {
+        it('when no cookie is available', () => {
+            expect(getConsentState()).toMatchObject({
+                guState: guDefaultState,
+                iabState: iabDefaultState,
+            });
+        });
+        it('when the euconsents cookie is available', () => {
+            readIabCookie.mockImplementation(() => iabMixedString);
+            readLegacyCookie.mockImplementation(() => legacyTrueCookie);
+
+            expect(getConsentState()).toMatchObject({
+                guState: guDefaultState,
+                iabState: iabMixedState,
+            });
+        });
+        it('when the GU_TK cookie is available', () => {
+            readLegacyCookie.mockImplementation(() => legacyTrueCookie);
+
+            expect(getConsentState()).toMatchObject({
+                guState: guDefaultState,
+                iabState: iabTrueState,
+            });
+        });
+        it('after a state change', () => {
+            setConsentState(guMixedState, iabMixedState);
```

Comment:
You could put an expectation checking that `getConsentState()` returns the default states before this `setConsentState` call, this would demonstrate the state has updated as expected when the second `getConsentState()` call occurs.